core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreUtil.java

```diff
@@ -111,4 +118,114 @@ public static Certificate getCertificate(String alias, KeyStore store) throws Ax
         }
     }
 
+    /**
+     * This method checks if the given key store name is for a custom key store by checking the prefix.
+     *
+     * @param keyStoreName      Custom key store name.
+     * @return Boolean value indicating if the given name is for a custom key store.
+     */
+    public static boolean isCustomKeyStore(String keyStoreName) {
+
+        return keyStoreName.startsWith(RegistryResources.SecurityManagement.CustomKeyStore.KEYSTORE_PREFIX);
+    }
+
+    /**
+     * This method builds the QName object for a configuration within the carbon.xml namespace.
+     *
+     * @param localPart     Local part of the configuration.
+     * @return QName object.
+     */
+    public static QName getQNameWithCarbonNS(String localPart) {
+
+        return new QName(ServerConstants.CARBON_SERVER_XML_NAMESPACE, localPart);
+    }
+
+    /**
+     * This method returns the OMElement object corresponding to the custom key store configuration in Carbon.xml file.
+     *
+     * @param keyStoreName          Name of the custom key store.
+     * @param serverConfiguration   ServerConfigurationService instance.
+     * @return OMElement object for the requested configuration.
+     * @throws CarbonException      If failed to retrieve the requested configuration.
+     */
+    public static OMElement getCustomKeyStoreConfigElement(
+            String keyStoreName, ServerConfigurationService serverConfiguration) throws CarbonException {
+
+        // Remove prefix if exists
```

Comment:
```suggestion
        // Remove prefix if exists.
```

Reply:
FIxed in https://github.com/wso2/carbon-kernel/pull/4025/commits/3aa02c8fd1024563e7aefe9329049af4802dfac3